app/services/teams-service.js

```diff
@@ -0,0 +1,295 @@
+    // Build the aggregation
+    const aggregation = [
+        { $sort: { 'name': 1 } },
+    ];
+
+    if (typeof options.search !== 'undefined') {
+        options.search = regexValidator.sanitizeRegex(options.search);
+        const match = { $match: { $or: [
+                    { 'name': { '$regex': options.search, '$options': 'i' }},
+                    { 'description': { '$regex': options.search, '$options': 'i' }},
+                ]}};
+        aggregation.push(match);
+    }
+
+    const facet = {
+        $facet: {
+            totalCount: [ { $count: 'totalCount' }],
+            documents: [ ]
+        }
+    };
+    if (options.offset) {
+        facet.$facet.documents.push({ $skip: options.offset });
+    }
+    else {
+        facet.$facet.documents.push({ $skip: 0 });
+    }
+    if (options.limit) {
+        facet.$facet.documents.push({ $limit: options.limit });
+    }
+    aggregation.push(facet);
+
+    // Retrieve the documents
+    Team.aggregate(aggregation, function(err, results) {
+        if (err) {
+            return callback(err);
+        }
+        else {
+            const teams = results[0].documents;
+            if (options.includePagination) {
+                let derivedTotalCount = 0;
+                if (results[0].totalCount.length > 0) {
+                    derivedTotalCount = results[0].totalCount[0].totalCount;
+                }
+                const returnValue = {
+                    pagination: {
+                        total: derivedTotalCount,
+                        offset: options.offset,
+                        limit: options.limit
+                    },
+                    data: teams
+                };
+                return callback(null, returnValue);
+            } else {
+                return callback(null, teams);
+            }
+        }
+    });
+};
+
+exports.retrieveById = function(teamId, callback) {
+    if (!teamId) {
+        const error = new Error(errors.missingParameter);
+        error.parameterName = 'teamId';
+        return callback(error);
+    }
+
+    Team.findOne({ 'id': teamId })
+        .lean()
+        .exec(function (err, team) {
+            if (err) {
+                if (err.name === 'CastError') {
+                    const error = new Error(errors.badlyFormattedParameter);
+                    error.parameterName = 'teamId';
+                    return callback(error);
+                } else {
+                    return callback(err);
+                }
+            } else {
+                return callback(null, team);
+            }
+        });
+};
+
+exports.createIsAsync = true;
+exports.create = async function(data) {
+    // Create the document
+    const team = new Team(data);
+
+    // Create a unique id for this user
+    // This should usually be undefined. It will only be defined when migrating user accounts from another system.
+    if (!team.id) {
+      team.id = `team--${uuid.v4()}`;
```

Comment:
Instead of using the prefix `team`, let's make the prefix `identity` (the same as for user accounts). This allows us to easily export this as a STIX identity object if we ever needed to. (STIX identities are flexible and can represent individuals, groups, or organizations by setting the `identity_class` so using the same prefix works fine.)